sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1185,6 +1188,36 @@ def _update_blocklist_on_error(
 
         return definitely_no_nodes_launched
 
+    # ---------------------------------- RunPod ---------------------------------- #
+    def _update_blocklist_on_runpod_error(
+            self, launchable_resources: 'resources_lib.Resources',
+            region: 'clouds.Region', zones: Optional[List['clouds.Zone']],
+            stdout: str, stderr: str):
+        del zones  # Unused.
+        style = colorama.Style
+        stdout_splits = stdout.split('\n')
+        stderr_splits = stderr.split('\n')
+        errors = [
+            s.strip()
+            for s in stdout_splits + stderr_splits
+            if 'RunPodError:' in s.strip()
+        ]
```

Comment:
We probably need to include the error in the comment [above](https://github.com/skypilot-org/skypilot/pull/2360#pullrequestreview-1610307505) in the errors. 
```suggestion
        errors = [
            s.strip()
            for s in stdout_splits + stderr_splits
            if any(e in s.strip() for e in ['runpod.error.QueryError:', 'RunPodError:'])
        ]
```